### ISSUE TITLE
On some projects, clicking "Run External Linter" gives false error about JDK configuration

### ISSUE THREAD

**Issue by AlyoshaVasilieva** [intellij-rust/intellij-rust#8324]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4324-213-nightly
* **Rust toolchain version:** 1.57.0 (f1edd0429 2021-11-29) x86_64-pc-windows-msvc
* **IDE name and version:** IntelliJ IDEA 2021.3 Ultimate Edition (IU-213.5744.223)
* **Operating system:** Windows 10 10.0
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
When clicking on the "Run External Linter" button to run Cargo Check/Clippy, IDEA says:
```
Error

The JDK is not configured properly for this project.
Inspection cannot proceed.
```

After clicking OK, a popup to select the JDK appears. It can be totally ignored as inspection then proceeds fine in the background.

## Steps to reproduce

1. File > New > Project from Version Control
2. Checkout `https://github.com/ngirard/lolcate-rs`
3. Wait for project to finish indexing, click "Run External Linter"

In my case, I eventually noticed that the project's IML has two weird things:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<module type="JAVA_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/target" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>
```

It is listed as a "JAVA_MODULE" and contains an entry for the JDK. Changing type to "RUST_MODULE" and removing "inheritedJdk" entry fixes the issue.

The source repo contains no IDEA files or Java code, IDEA is generating this erroneously for some reason. Maybe this is a bug in IDEA rather than this plugin, I'm not sure. If possible though the Rust plugin shouldn't be prompting me to set the JDK as it is unrelated to Cargo Check/Clippy.

**Comment by sebr74** [comment]
> Changing type to "RUST_MODULE" and removing "inheritedJdk" entry fixes the issue.
I was also affected by this bug. Thanks for providing the workaround. I confirmed that this made it work for me
